src/sidebar/services/test/toast-messenger-test.js

```diff
@@ -84,6 +84,17 @@ describe('ToastMessengerService', () => {
       assert.calledOnce(fakeStore.getToastMessage);
       assert.notCalled(fakeStore.updateToastMessage);
     });
+
+    it('emits "toastMessageAdded" event', () => {
+      fakeStore.hasToastMessage.returns(false);
+
+      const fakeHandler = sinon.stub();
+      service.on('toastMessageAdded', fakeHandler);
+
+      service.success('hooray', {});
+
+      assert.called(fakeHandler);
```

Comment:
I'd suggest to use `calledWith` here and check the arguments as well.

Reply:
That was my initial intention, but it was a bit tricky because part of the message object is generated internally (mainly the `id`).

With `jest` I would do something like `expect.objectContaining(...)` and assert on a partial object. I couldn't find an equivalent with our stack, but I'll check a bit further.

Reply:
Scratch that. `sinon.match` is what I need and is already extensively used on this very test 🤦🏼

Reply:
The equivalent in our codebase is [Sinon matchers](https://sinonjs.org/releases/v15/matchers/) (and [assertions](https://sinonjs.org/releases/v15/assertions/)). Regex search for `calledWith.*match` and `assert\.match` for some examples.

Reply:
> Scratch that. sinon.match is what I need and is already extensively used on this very test 🤦🏼

Ah, you found it while I was writing.